### PR TITLE
display player names

### PR DESCRIPTION
**Pull request by rom1504**

### CODE REVIEW

File: viewer/lib/worldView.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ class WorldView extends EventEmitter {
       // 'move': botPosition,
       entitySpawn: function (e) {
         const type = e.type === 'mob' ? e.name.toLowerCase() : e.type
-        worldView.emitter.emit('entity', { id: e.id, type, pos: e.position, width: e.width, height: e.height })
+        worldView.emitter.emit('entity', { id: e.id, type, pos: e.position, width: e.width, height: e.height, username: e.username })
```
Comment (Karang) [inline viewer/lib/worldView.js:19]: this need to be added also L54

File: viewer/lib/entities.js [hunk 2]
```diff
@@ -13,6 +13,29 @@ function getEntityMesh (entity, scene) {
   } else if (entity.type) {
     try {
       const e = new Entity('1.16.4', entity.type, scene)
+
+      if ((entity.type === 'player')) {
```
Comment (Karang) [inline viewer/lib/entities.js:17]: what about just checking if username is defined ?

**Comment by rom1504** [comment]
potential follow up : 
* use these fonts https://github.com/PrismarineJS/prismarine-web-client/tree/master/assets
* better name centering
* non player named entities

File: viewer/lib/entities.js [hunk 3]
```diff
@@ -13,6 +13,29 @@ function getEntityMesh (entity, scene) {
...
+        ctx.textBaseline = 'top'
+
+        const txt = entity.username
+        ctx.fillText(txt, 100, 0)
+
+        const tex = new THREE.Texture(canvas)
+        tex.needsUpdate = true
+        const spriteMat = new THREE.SpriteMaterial({ map: tex })
+        const sprite = new THREE.Sprite(spriteMat)
+        sprite.position.y += 2.2
```
Comment (Karang) [inline viewer/lib/entities.js:35]: `entity.height + 0.4` ?

**Review by Karang — CHANGES_REQUESTED** [review]

File: viewer/lib/entities.js (hunk 3 above)
Comment (rom1504) [reply viewer/lib/entities.js]: done

File: viewer/lib/entities.js (hunk 2 above)
Comment (rom1504) [reply viewer/lib/entities.js]: done

File: viewer/lib/worldView.js (hunk 1 above)
Comment (rom1504) [reply viewer/lib/worldView.js:19]: done

**Review by Karang — APPROVED** [review]

**Pull request merged**